subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/matcher/TextMatchers.kt

```diff
@@ -0,0 +1,144 @@
+    private val method: TextViewMethod,
+) : BoundedDiagnosingMatcher<View, TextView>(TextView::class.java) {
+    private var resourceName: String? = null
+    private var expectedText: String? = null
+
+    internal enum class TextViewMethod {
+        GET_TEXT, GET_HINT
+    }
+
+    override fun describeMoreTo(description: Description) {
+        when (method) {
+            TextViewMethod.GET_TEXT -> description.appendText("view.getText()")
+            TextViewMethod.GET_HINT -> description.appendText("view.getHint()")
+        }
+        description.appendText(" equals string from resource id: ").appendValue(resourceId)
+        if (null != resourceName) {
+            description.appendText(" [").appendText(resourceName).appendText("]")
+        }
+        if (null != expectedText) {
+            description.appendText(" value: ").appendText(expectedText!!.highlightNbsp())
+        }
+    }
+
+    override fun matchesSafely(textView: TextView, mismatchDescription: Description): Boolean {
+        if (null == expectedText) {
+            try {
+                expectedText = textView.resources.getString(resourceId)
+            } catch (ignored: Resources.NotFoundException) {
+                /* view could be from a context unaware of the resource id. */
+            }
+            resourceName = safeGetResourceEntryName(
+                textView.resources,
+                resourceId
+            )
+        }
+        val expectedText: String? = expectedText
+        val actualText: String = when (method) {
+            TextViewMethod.GET_TEXT -> {
+                mismatchDescription.appendText("view.getText() was ")
+                textView.text.toString()
+            }
+            TextViewMethod.GET_HINT -> {
+                mismatchDescription.appendText("view.getHint() was ")
+                textView.hint.toString()
+            }
+        }
+        mismatchDescription.appendValue(actualText.highlightNbsp())
+        // FYI: actualText may not be string ... its just a char sequence convert to string.
+        expectedText ?: return false
+        return expectedText == actualText
+    }
+
+    private fun safeGetResourceEntryName(res: Resources, id: Int): String? {
+        return try {
+            if (isViewIdGenerated(id)) null else res.getResourceEntryName(id)
+        } catch (e: Resources.NotFoundException) {
+            null
+        }
+    }
+
+    /**
+     * IDs generated by [View.generateViewId] will fail if used as a resource ID in attempted
+     * resources lookups. This now logs an error in API 28, causing test failures. This method is
+     * taken from [View.isViewIdGenerated] to prevent resource lookup to check if a view id was
+     * generated.
+     */
+    private fun isViewIdGenerated(id: Int): Boolean {
+        return id and -0x1000000 == 0 && id and 0x00FFFFFF != 0
+    }
+}
+
+internal fun String.highlightNbsp() = replace(NBSP_SYMBOL, "[NBSP]")
+
+private const val NBSP_SYMBOL = " "
```

Comment:
```suggestion
private const val NBSP_SYMBOL = "\u00A0"
```
To make it more obvious.

Reply:
Fixed